go/sqltypes/type.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sqltypes
+
+import (
+	"fmt"
+
+	"github.com/youtube/vitess/go/vt/proto/query"
+)
+
+// These bit flags can be used to query on the
+// common properties of types.
+const (
+	IsNumber   = int(query.Flag_ISNUMBER)
+	IsUnsigned = int(query.Flag_ISUNSIGNED)
+	IsFloat    = int(query.Flag_ISFLOAT)
+	IsQuoted   = int(query.Flag_ISQUOTED)
+	IsText     = int(query.Flag_ISTEXT)
+	IsBinary   = int(query.Flag_ISBINARY)
+)
+
+// Vitess data types. These are idiomatically
+// named synonyms for the query.Type values.
+const (
+	Null      = query.Type_NULL
+	Int8      = query.Type_INT8
+	Uint8     = query.Type_UINT8
+	Int16     = query.Type_INT16
+	Uint16    = query.Type_UINT16
+	Int24     = query.Type_INT24
+	Uint24    = query.Type_UINT24
+	Int32     = query.Type_INT32
+	Uint32    = query.Type_UINT32
+	Int64     = query.Type_INT64
+	Uint64    = query.Type_UINT64
+	Float     = query.Type_FLOAT
+	Double    = query.Type_DOUBLE
+	Timestamp = query.Type_TIMESTAMP
+	Date      = query.Type_DATE
+	Time      = query.Type_TIME
+	Datetime  = query.Type_DATETIME
+	Year      = query.Type_YEAR
+	Decimal   = query.Type_DECIMAL
+	Text      = query.Type_TEXT
+	Blob      = query.Type_BLOB
+	VarChar   = query.Type_VARCHAR
+	VarBinary = query.Type_VARBINARY
+	Char      = query.Type_CHAR
+	Binary    = query.Type_BINARY
+	Bit       = query.Type_BIT
+	Enum      = query.Type_ENUM
+	Set       = query.Type_SET
+)
+
+// bit-shift the mysql flags by one byte so we
+// can merge them with the mysql types.
+const (
+	mysqlUnsigned = 32 << 8
+	mysqlBinary   = 128 << 8
+	mysqlEnum     = 256 << 8
+	mysqlSet      = 2048 << 8
+
+	relevantFlags = mysqlUnsigned |
+		mysqlBinary |
+		mysqlEnum |
+		mysqlSet
+)
+
+// If you add to this map, make sure you add a test case
+// in tabletserver/endtoend.
+var mysqlToType = map[int64]query.Type{
+	1:   Int8,
+	2:   Int16,
+	3:   Int32,
+	4:   Float,
+	5:   Double,
+	6:   Null,
+	7:   Timestamp,
+	8:   Int64,
+	9:   Int24,
+	10:  Date,
+	11:  Time,
+	12:  Datetime,
+	13:  Year,
+	16:  Bit,
+	246: Decimal,
+	252: Text,
+	253: VarChar,
+	254: Char,
+}
+
+var modifier = map[int64]query.Type{
+	int64(Int8) | mysqlUnsigned:  Uint8,
+	int64(Int16) | mysqlUnsigned: Uint16,
+	int64(Int32) | mysqlUnsigned: Uint32,
+	int64(Int64) | mysqlUnsigned: Uint64,
+	int64(Int24) | mysqlUnsigned: Uint24,
+	int64(Text) | mysqlBinary:    Blob,
+	int64(VarChar) | mysqlBinary: VarBinary,
+	int64(Char) | mysqlBinary:    Binary,
+	int64(Char) | mysqlEnum:      Enum,
+	int64(Char) | mysqlSet:       Set,
+}
+
+// typeToMySQL is the reverse of mysqlToType.
+var typeToMySQL = map[query.Type]struct {
+	typ   int64
```

Comment:
"typ" is the German translation for "type". So it reads just fine for me :)

(Since it's a noun, you would capitalize it in German though: "Typ" ;)